public/friendlycode/js/fc/filesystem-sync.js

```diff
@@ -0,0 +1,104 @@
+define(["jquery"], function($) {
+  "use strict";
+
+  var FileSystemSync = {};
+
+  function pushFileChange(url, csrfToken, fs, path) {
+    var options = {
+      contentType: "application/json",
+      headers: {
+        "X-Csrf-Token": csrfToken
+      },
+      type: "PUT",
+      url: url
+    };
+
+    function send() {
+      var request = $.ajax(options);
+      request.done(function() {
+        if(request.readyState !== 4) {
+          return;
+        }
+
+        if(request.status !== 201 && request.status !== 200) {
+          // TODO: handle error case here
+          console.error("Server did not persist file");
+          return;
+        }
+
+        console.log("Successfully persisted ", path);
+      });
+      request.fail(function(jqXHR, status, err) {
+        console.error("Failed to send request to persist the file to the server with: ", err);
+      });
+    }
+
+    fs.readFile(path, function(err, data) {
+      if(err) {
+        // TODO: handle errors
+        throw err;
```

Comment:
Can we `console.error("[Bramble]...", err);` here instead, so that it looks like we've dealt with it on the console vs. just blowing up?  Also, how bad an error is this, since it would just mean that somehow we thought we needed to deal with this file, and actually we don't (i.e., it's not there now).  Maybe we can just `console.warn` and bail here.